src/transaction.ts

```diff
@@ -19,647 +31,599 @@ const TRANSACTION_HASH_LENGTH = 32;
-      value: new BigNumber(transaction.value.toString()).toFixed(0),
-      data: new TransactionPayload(Buffer.from(transaction.data)),
-      nonce: Number(transaction.nonce),
-      gasPrice: Number(transaction.gasPrice),
-      receiverUsername: transaction.receiverUsername,
-      senderUsername: transaction.senderUsername,
-      options: transaction.options,
-      version: transaction.version
-    });
-
-    if (transaction.guardian) {
-      tx.guardian = Address.fromBech32(transaction.guardian)
-    }
-
-    if (transaction.signature.length) {
-      tx.applySignature(transaction.signature);
-    }
-
-    if (transaction.guardianSignature.length) {
-      tx.applyGuardianSignature(transaction.guardianSignature);
-    }
-
-    return tx;
-  }
+    /**
+     * The nonce of the transaction (the account sequence number of the sender).
+     */
+    public nonce: bigint;
+
+    /**
+     * The value to transfer.
+     */
+    public value: bigint;
+
+    /**
+     * The address of the sender.
+     */
+    public sender: string;
+
+    /**
+     * The address of the receiver.
+     */
+    public receiver: string;
+
+    /**
+     * The username of the sender.
+     */
+    public senderUsername: string;
+
+    /**
+     * The username of the receiver.
+     */
+    public receiverUsername: string;
+
+    /**
+     * The gas price to be used.
+     */
+    public gasPrice: bigint;
+
+    /**
+     * The maximum amount of gas to be consumed when processing the transaction.
+     */
+    public gasLimit: bigint;
+
+    /**
+     * The payload of the transaction.
+     */
+    public data: Uint8Array;
+
+    /**
+     * The chain ID of the Network (e.g. "1" for Mainnet).
+     */
+    public chainID: string;
+
+    /**
+     * The version, required by the Network in order to correctly interpret the contents of the transaction.
+     */
+    public version: number;
+
+    /**
+     * The options field, useful for describing different settings available for transactions.
+     */
+    public options: number;
+
+    /**
+     * The address of the guardian.
+     */
+    public guardian: string;
+
+    /**
+     * The signature.
+     */
+    public signature: Uint8Array;
+
+    /**
+     * The signature of the guardian.
+     */
+    public guardianSignature: Uint8Array;
+
+    /**
+     * The transaction hash, also used as a transaction identifier.
+     */
+    private hash: TransactionHash;
+
+    /**
+     * Creates a new Transaction object.
+     */
+    public constructor({
+        nonce,
+        value,
+        sender,
+        receiver,
+        senderUsername,
+        receiverUsername,
+        gasPrice,
+        gasLimit,
+        data,
+        chainID,
+        version,
+        options,
+        guardian,
+        signature,
+        guardianSignature,
+    }: {
+        nonce?: INonce | bigint;
+        value?: ITransactionValue | bigint;
+        sender?: IAddress | string;
+        receiver?: IAddress | string;
```

Comment:
shouldn't we have mandatory fields on the constructor? Like sender, receiver, gas limit and chain?

Reply:
Indeed, fixed. Reverted, and now we have the same required fields as in PY (as in JS v12, as well): sender, receiver, chain ID and gas limit.